redaxo/src/core/lib/util/validator.php

```diff
@@ -53,6 +53,24 @@ public function add($type, $message = null, $option = null)
         return $this;
     }
 
+    /**
+     * Returns whether the validator contains a validation rule for the given type.
+     *
+     * @param string $type
+     */
+    public function contains($type): bool
```

Comment:
Cool wäre eigentlich, wenn zum Beispiel auch das `maxlength`-Attribute automatisch anhand der Validierung gesetzt werden würde.
Dann reicht aber eine `contains`-Methode nicht, sondern wir müssten auch noch an den zugehörigen Wert kommen.

Daher vielleicht eher in rex_form per foreach alle Validierungen durchlaufen, und wenn notEmpty, dann required, und wenn maxLength, dann entsprechen maxlength-attr.

Reply:
ok, d.h. direkt im rex_validator einen Iterator implementieren oder ne separate methode die einen iterator returned ?

Reply:
Zum beispiel. Oder einfach das Array liefern.
Oder man macht ne art getter-methode: `$validator->get('notEmpty')`. Da würde dann rex_form doch nicht schleife durchlaufen, sondern explizit die abfragen, die für rex_form interessant sind.
wenn `null`, dann nicht vorhanden.
Fraglich ist, was passiert, wenn es mehrere `maxLength` gäbe.

So oder so wäre dann ganz cool kleine Objekte zu haben, die das Triple Typ, Message und Option halten. Also `$this->types` wäre dann ein Array von Objekten dieser Klasse.